elichika/elichika/parser/values.py

```diff
@@ -258,7 +258,8 @@ def get_outputs(self) -> 'List[FieldOutput]':
         ret = []
 
         for key, att in self.attributes.items():
-
+            if att.name in ['shape', 'size']:
```

Comment:
I think this implementation causes error as follows (it is very rare case).

class A:
    def __init__(self):
        self.size = 'Size'

size, shape is regarded as bultin property.
I implemented shape in NDArray

https://github.com/pfnet-research/chainer-compiler/blob/master/elichika/elichika/parser/functions_ndarray.py#L126

it is redundant. But it can support many cases.
Do you change it?

Reply:
I wasn't aware that there was an implementation to handle NDArrayFunctions. In that case, there is a bug in the implementation then. It doesn't handle the following case.

```python
xs = np.random.rand(3,4,5)
[x.shape for x in xs]
```

I'm sure your implementation can be extended to fix this though.

Reply:
Thank you!